### PR TITLE
50 Update workevents

### LINKED ISSUE #50: Update workevents
Add the feature of updating a workevent's data.
Add the features of listing the possible workers to assign to a workevent in it's time range.

### PR DESCRIPTION
**Pull request by akosgdcsi**
love the gitkraken

### CODE REVIEW

File: wman-backend/Wman.Logic/Classes/EventLogic.cs [hunk 1]
```diff
@@ -226,5 +252,41 @@ public async Task DnDEvent(int Id, DnDEventDTO newWorkEvent)
                 throw new ArgumentException("Events are not at the same day or start is after the finishing date");
             }
         }
+        private async Task<bool> WorkerTimeCheck(List<WmanUser> assignedUsers, DateTime startDate , DateTime finishDate)
+        {
+            if (assignedUsers.Count > 0)
+            {
+                var eventsAtUpdateTime = await(from x in eventRepo.GetAll()
+                                             where x.EstimatedStartDate >= startDate && x.EstimatedStartDate <= finishDate && x.EstimatedFinishDate >= startDate && x.EstimatedFinishDate <= finishDate
```
Comment (pop-simon) [inline wman-backend/Wman.Logic/Classes/EventLogic.cs]: This condition fails badly. Here is an example:
**Before update**
``` json
{
    "id": 2,
    "jobDescription": "Example event #2",
    "estimatedStartDate": "2021-11-10T05:00:00",
    "estimatedFinishDate": "2021-11-10T06:30:00",
    "assignedUsers": [
      {
        "username": "user1",
        "email": "pelda@abc.com",
        "firstname": "First",
        "lastname": "User",
        "profilePicture": null
      }
    ],
    "labels": [],
    "address": {
      "city": "Budapest",
      "street": "Doberdó út",
      "zipCode": 1034,
      "buildingNumber": "6/A.",
      "floorDoor": null
    },
    "workStartDate": "2021-11-10T05:07:00",
    "workFinishDate": "0001-01-01T00:00:00",
    "status": "started"
  },
  {
    "id": 1,
    "jobDescription": "Example event #1",
    "estimatedStartDate": "2021-11-11T03:00:00",
    "estimatedFinishDate": "2021-11-11T04:00:00",
    "assignedUsers": [
      {
        "username": "user1",
        "email": "pelda@abc.com",
        "firstname": "First",
        "lastname": "User",
        "profilePicture": null
      }
    ],
    "labels": [],
    "address": {
      "city": "Budapest",
      "street": "Bécsi út",
      "zipCode": 1034,
      "buildingNumber": "104-108.",
      "floorDoor": null
    },
    "workStartDate": "0001-01-01T00:00:00",
    "workFinishDate": "0001-01-01T00:00:00",
    "status": "awaiting"
  }
```

**After update**
``` json
{
    "id": 2,
    "jobDescription": "Example event #2",
    "estimatedStartDate": "2021-11-10T05:00:00",
    "estimatedFinishDate": "2021-11-10T06:30:00",
    "assignedUsers": [
      {
        "username": "user1",
        "email": "pelda@abc.com",
        "firstname": "First",
        "lastname": "User",
        "profilePicture": null
      }
    ],
    "labels": [],
    "address": {
      "city": "Budapest",
      "street": "Doberdó út",
      "zipCode": 1034,
      "buildingNumber": "6/A.",
      "floorDoor": null
    },
    "workStartDate": "2021-11-10T05:07:00",
    "workFinishDate": "0001-01-01T00:00:00",
    "status": "started"
  },
  {
    "id": 1,
    "jobDescription": "string",
    "estimatedStartDate": "2021-11-10T05:30:00",
    "estimatedFinishDate": "2021-11-10T07:30:00",
    "assignedUsers": [
      {
        "username": "user1",
        "email": "pelda@abc.com",
        "firstname": "First",
        "lastname": "User",
        "profilePicture": null
      }
    ],
    "labels": [],
    "address": {
      "city": "string",
      "street": "string",
      "zipCode": 1111,
      "buildingNumber": "string",
      "floorDoor": "string"
    },
    "workStartDate": "0001-01-01T00:00:00",
    "workFinishDate": "0001-01-01T00:00:00",
    "status": "awaiting"
  }
```
In this case the worker should be at two places from 5:30 to 6:30. Please update the condition as suggested and test a few scenarios.
```suggestion
                                             where x.EstimatedStartDate <= finishDate && x.EstimatedFinishDate >= startDate
```

**Review by pop-simon — CHANGES_REQUESTED** [review]
The issue's description is the following:

> Add the feature of updating a workevent's data.
Add the features of listing the possible workers to assign to a workevent in it's time range.

**I am afraid that the request for the second feature has not been implemented.**

**Comment by akosgdcsi** [comment]
Bujakasha, srry that I didnt see the full issue discription😿 I think the 2nd part works fine, but pls check 🉐

**Review by tsaskater — APPROVED** [review]
Works fine

File: wman-backend/Wman.Logic/Classes/AllInWorkEventLogic.cs [hunk 2]
```diff
@@ -31,5 +35,63 @@ public async Task<WorkEventForWorkCardDTO> ForWorkCard(int eventId)
...
+            
+
+            if (fromDate<toDate)
+            {
+                var allUsers = userManager.Users
+                .Include(y => y.WorkEvents)
+                .Include(z => z.ProfilePicture);
+
+                var notavailableWorkers = await (from x in workEvent.GetAll()
+                                              where fromDate <= x.EstimatedStartDate && toDate >= x.EstimatedFinishDate
```
Comment (pop-simon) [inline wman-backend/Wman.Logic/Classes/AllInWorkEventLogic.cs]: **You made the same mistake as previously.**
The event:
``` json
{
    "id": 4,
    "jobDescription": "Example event #4",
    "estimatedStartDate": "2021-11-17T05:00:00",
    "estimatedFinishDate": "2021-11-17T09:00:00",
    "assignedUsers": [
      {
        "username": "user1",
        "email": "pelda@abc.com",
        "firstname": "First",
        "lastname": "User",
        "profilePicture": null
      },
      {
        "username": "user2",
        "email": "Pelda2@gmail.com",
        "firstname": "Second",
        "lastname": "User",
        "profilePicture": null
      }
    ], etc.
```

The parameters that I used for testing:
`fromDate: 2021-11-17T08:00:00`
`toDate: 2021-11-17T12:00:00`

The method's result you just implemented:
``` json
[
  {
    "username": "user1",
    "email": "pelda@abc.com",
    "firstname": "First",
    "lastname": "User",
    "profilePicture": null
  },
  {
    "username": "user2",
    "email": "Pelda2@gmail.com",
    "firstname": "Second",
    "lastname": "User",
    "profilePicture": null
  },
  {
    "username": "user3",
    "email": "Pelda3@gmail.com",
    "firstname": "Third",
    "lastname": "User",
    "profilePicture": null
  }
]
```

```suggestion
                                              where fromDate <= x.EstimatedFinishDate && toDate >= x.EstimatedStartDate
```

**Review by pop-simon — CHANGES_REQUESTED** [review]

**Comment by pop-simon** [comment]
Furthermore, the branch #53 has been merged to development so please pull then merge the current development branch into this issues branch and implement the proper exception handling related to this issue. @akosgdcsi ☠
***The merge should not cause conflict.***

**Review by pop-simon — APPROVED** [review]
Seems good, well done 👀

**Pull request merged**